File: leetcode1503/solution.rb
# @param {Integer} n
# @param {Integer[]} left
# @param {Integer[]} right
# @return {Integer}
def get_last_moment(n, left, right)
    if (if left.empty? then 0 else left.max end) > (if right.empty? then 0 else n - right.min end) then (if left.empty? then 0 else left.max end) else (if right.empty? then 0 else n - right.min end) end
end

File: leetcode1503/solution.exs
defmodule Solution do
  @spec get_last_moment(n :: integer, left :: [integer], right :: [integer]) :: integer
  def get_last_moment(n, left, right) do
    case Enum.empty?(left) do
        true -> 0
        false -> Enum.max(left)
        end
    |> (&(case Enum.empty?(right) do
        true -> [&1, 0]
        false -> [&1, n - Enum.min(right)]
        end)).()
    |> Enum.max
  end
end
